### ISSUE TITLE
password validation policy is breaking existing additional users 

### ISSUE THREAD

**Issue by Anand-chenepalli** [terraform-google-modules/terraform-google-sql-db#464]
### TL;DR

When trying to use password validation policy, additional existing users password is getting changed as random_password resource arguments (min_special , special  and length)  causing force replacement. If  min_special , special  and length arguments are added to below lifecycle then it will not break existing additional users.

lifecycle {
    ignore_changes = [
      min_lower, min_upper, min_numeric
    ]
  }

### Expected behavior

password validation policy will be applied on new additional users without any issue and existing users will not impact.

### Observed behavior

module.pgsql-db.module.gcp-pgsql.random_password.additional_passwords["adduser2"] must be replaced
-/+ resource "random_password" "additional_passwords" {
      ~ bcrypt_hash = (sensitive value)
      ~ id          = "none" -> (known after apply)
      ~ min_lower   = 0 -> 1
      ~ min_numeric = 0 -> 1
      ~ min_special = 0 -> 1 # forces replacement
      ~ min_upper   = 0 -> 1
      ~ result      = (sensitive value)
      ~ special     = false -> true # forces replacement
        # (6 unchanged attributes hidden)
    }

  module.pgsql-db.module.gcp-pgsql.random_password.additional_passwords["adduser3"] must be replaced
-/+ resource "random_password" "additional_passwords" {
      ~ bcrypt_hash = (sensitive value)
      ~ id          = "none" -> (known after apply)
      ~ min_lower   = 0 -> 1
      ~ min_numeric = 0 -> 1
      ~ min_special = 0 -> 1 # forces replacement
      ~ min_upper   = 0 -> 1
      ~ result      = (sensitive value)
      ~ special     = false -> true # forces replacement
        # (6 unchanged attributes hidden)
    }

   module.pgsql-db.module.gcp-pgsql.random_password.additional_passwords["adduser6"] must be replaced
-/+ resource "random_password" "additional_passwords" {
      ~ bcrypt_hash = (sensitive value)
      ~ id          = "none" -> (known after apply)
      ~ min_lower   = 0 -> 1
      ~ min_numeric = 0 -> 1
      ~ min_special = 0 -> 1 # forces replacement
      ~ min_upper   = 0 -> 1
      ~ result      = (sensitive value)
      ~ special     = false -> true # forces replacement
        # (6 unchanged attributes hidden)
    }

   module.pgsql-db.module.gcp-pgsql.random_password.additional_passwords["adduser7"] will be created
  + resource "random_password" "additional_passwords" {
      + bcrypt_hash = (sensitive value)
      + id          = (known after apply)
      + keepers     = {
          + "name" = "pgsql-cftv15-test-06ff322e"
        }
      + length      = 32
      + lower       = true
      + min_lower   = 1
      + min_numeric = 1
      + min_special = 1
      + min_upper   = 1
      + number      = true
      + numeric     = true
      + result      = (sensitive value)
      + special     = true
      + upper       = true
    }

 module.pgsql-db.module.gcp-pgsql.random_password.user-password must be replaced
-/+ resource "random_password" "user-password" {
      ~ bcrypt_hash = (sensitive value)
      ~ id          = "none" -> (known after apply)
      ~ min_lower   = 0 -> 1
      ~ min_numeric = 0 -> 1
      ~ min_special = 0 -> 1 # forces replacement
      ~ min_upper   = 0 -> 1
      ~ result      = (sensitive value)
      ~ special     = false -> true # forces replacement
        # (6 unchanged attributes hidden)

### Terraform Configuration

```hcl
additional_users = [{
    name            = "adduser3"
    password        = null
    random_password = true
    },
    {
      name            = "adduser2"
      password        = null
      random_password = true
    },
    {
      name            = "adduser6"
      password        = null
      random_password = true
    },
    {
      name            = "adduser7"
      password        = null
      random_password = true
    }
  ]
```

### Terraform Version

```sh
Terraform v1.3.3
on windows_amd64
+ provider registry.terraform.io/cyrilgdn/postgresql v1.16.0
+ provider registry.terraform.io/hashicorp/google v4.63.1
+ provider registry.terraform.io/hashicorp/google-beta v4.63.1
+ provider registry.terraform.io/hashicorp/null v3.2.1
+ provider registry.terraform.io/hashicorp/random v3.5.1

Your version of Terraform is out of date! The latest version
is 1.4.6. You can update by downloading from https://www.terraform.io/downloads.html
```

**Comment by bharathkkb** [comment]
@Anand-chenepalli I believe this is related to https://github.com/terraform-google-modules/terraform-google-sql-db/pull/461?

**Comment by Anand-chenepalli** [comment]
@bharathkkb Yes you are right

**Comment by imrannayer** [comment]
Fixed in #461
